agent/agent_test.go

```diff
@@ -1982,6 +1982,72 @@ func TestAgent_loadServices_token(t *testing.T) {
 	}
 }
 
+func TestAgent_loadServices_sidecar(t *testing.T) {
+	t.Parallel()
+	a := NewTestAgent(t.Name(), `
+		service = {
+			id = "rabbitmq"
+			name = "rabbitmq"
+			port = 5672
+			token = "abc123"
+			connect = {
+				sidecar_service {}
+			}
+		}
+	`)
+	defer a.Shutdown()
+
+	services := a.State.Services()
+	if _, ok := services["rabbitmq"]; !ok {
+		t.Fatalf("missing service")
```

Comment:
Nitpick but should these busing `require` helpers?

Reply:
Yeah could do, it was copy-paste from other similar code I think.